ezpublish_legacy/ezsylius/login_handler/ezsyliususer.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+class eZSyliusUser extends eZUser
+{
+    /**
+     * Checks if there's an admin sylius user having login entered by the user.
+     * If so, it checks password entered is correct and if that is the case,
+     * it returns the legacy user.
+     */
+    public static function loginUser($login, $password, $authenticationMatch = false)
+    {
+        $container = ezpKernel::instance()->getServiceContainer();
+        try {
+            /** @var \Netgen\Bundle\EzSyliusBundle\Entity\AdminUser $syliusUser */
+            $syliusUser = $container->get('netgen_ez_sylius.admin_user.provider.email_or_name_based')
```

Comment:
Could we do this in the class constructor? Something like this: https://github.com/netgen/NetgenEzSyliusBundle/blob/master/ezpublish_legacy/ezsylius/datatypes/syliusproduct/syliusproducttype.php#L36-L40. Encoder factory too?